utils/snapshot_extractor.mjs

```diff
@@ -54,13 +54,16 @@ export default async function snapshotExtractor(extractor, { inputs }) {
   const ret = await extractor.init(...inputs);
   if (ret.write) write(ret.write);
 
-  prepareMessages(ret.messages, extractor.name)
-    .filter(({ type }) => type !== "extraction" && type !== "transformation")
-    .forEach((message) => {
+  const preparedMessages = prepareMessages(ret.messages, extractor.name).filter(
+    ({ type }) => type !== "extraction" && type !== "transformation"
+  );
+
+  if (preparedMessages.length) {
+    preparedMessages.forEach((message) => {
       postMessage(message);
     });
-
-  await once(worker, "exit");
+    await once(worker, "exit");
+  }
 
```

Comment:
Fixed a bug that if `init` returns `messages: []` the `snapshotExtractor` doesn't close.

Reply:
should have been a separate pr and an issue

Reply:
https://github.com/neume-network/strategies/commit/9ee5c6d2e4b9f8ac734c4650fc7226c1279c1a22